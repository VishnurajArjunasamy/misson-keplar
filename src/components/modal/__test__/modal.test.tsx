import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import Modal from "../modal";

const mockStore = configureStore([]);

describe("Model  Component", () => {
  let store: ReturnType<typeof mockStore>;

  beforeEach(() => {
    store = mockStore({
      sideBar: {
        isDarkMode: false,
      },
    });
  });

  it("renders modal with dark mode styles", () => {
    store = mockStore({
      sideBar: {
        isDarkMode: true,
      },
    });
    const { asFragment } = render(
      <Provider store={store}>
        <Modal>
          <div>Content</div>
        </Modal>
      </Provider>
    );
    expect(screen.getByText("Content").parentNode?.parentNode).toHaveClass(
      "dark"
    );
    expect(screen.getByText("Content").parentNode?.parentNode).not.toHaveClass(
      "light"
    );

    expect(asFragment()).toMatchSnapshot();
  });

  jest.useFakeTimers();
  test("calls closeModal when background is clicked", () => {
    const closeModalMock = jest.fn();
    const { asFragment } = render(
      <Provider store={store}>
        <Modal closeModal={closeModalMock}>
          <div>Content</div>
        </Modal>
      </Provider>
    );

    const backgroundDiv = screen.getByText("Content").parentNode?.parentNode;
    if (backgroundDiv) {
      fireEvent.click(backgroundDiv);
    }
    jest.advanceTimersByTime(500);
    expect(closeModalMock).toHaveBeenCalledTimes(1);

    expect(asFragment()).toMatchSnapshot();
  });

  it("does not call closeModal when content is clicked", () => {
    const closeModalMock = jest.fn();
    const { asFragment } = render(
      <Provider store={store}>
        <Modal closeModal={closeModalMock}>
          <div>Content</div>
        </Modal>
      </Provider>
    );

    const contentDiv = screen.getByText("Content");
    fireEvent.click(contentDiv);
    expect(closeModalMock).toHaveBeenCalledTimes(0);

    expect(asFragment()).toMatchSnapshot();
  });
});
