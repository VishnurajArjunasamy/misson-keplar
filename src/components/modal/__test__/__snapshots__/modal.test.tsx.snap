// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member Card Component calls closeModal when background is clicked 1`] = `
<DocumentFragment>
  <div
    class="modal light"
  >
    <div
      class="content"
    >
      <div>
        Content
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Member Card Component does not call closeModal when content is clicked 1`] = `
<DocumentFragment>
  <div
    class="modal light"
  >
    <div
      class="content slideIn "
    >
      <div>
        Content
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Member Card Component renders modal with dark mode styles 1`] = `
<DocumentFragment>
  <div
    class="modal dark"
  >
    <div
      class="content slideIn "
    >
      <div>
        Content
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Model  Component calls closeModal when background is clicked 1`] = `
<DocumentFragment>
  <div
    class="modal light"
  >
    <div
      class="content slideOut "
    >
      <div>
        Content
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Model  Component does not call closeModal when content is clicked 1`] = `
<DocumentFragment>
  <div
    class="modal light"
  >
    <div
      class="content slideIn "
    >
      <div>
        Content
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Model  Component renders modal with dark mode styles 1`] = `
<DocumentFragment>
  <div
    class="modal dark"
  >
    <div
      class="content slideIn "
    >
      <div>
        Content
      </div>
    </div>
  </div>
</DocumentFragment>
`;
